runner/src/main/kotlin/com/cosmotech/runner/service/RunnerService.kt

```diff
@@ -297,5 +279,24 @@ class RunnerService(
       val rbacSecurity = csmRbac.setDefault(this.getRbacSecurity(), role, this.roleDefinition)
       this.setRbacSecurity(rbacSecurity)
     }
+
+    fun delete() {
```

Comment:
The delete should be in the runnerService class. runnerService Class do action on runnerInstance Class. runnerInstance should only provide status.